### PR TITLE
Added maddy example config

### PR DESCRIPTION
**Pull request by elmodor**
Added an example configuration for the [Maddy Mail Server](https://maddy.email/)

### CODE REVIEW

**Comment by elmodor** [comment]
Please hold, I found an issue with different mail clients. Will adjust the config to be more flexible

File: example_configs/maddy.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Documentation for maddy LDAP can be found [here](https://maddy.email/reference/auth/ldap/).  
+Maddy will automatically create an imap-acct if a new user connects via LDAP.
+
+### URL
+Change to your LLDAP URL
+```
+urls ldap://lldap.example.com:3890
+```
+
+#### Docker
```
Comment (nitnelave) [inline example_configs/maddy.md:12]: No need for this section. People should know what the URL for their LLDAP server is, at least.

**Review by nitnelave — APPROVED** [review]

**Comment by elmodor** [comment]
Since different mail clients seem to send different usernames to maddy (which then uses these for the LDAP auth) I extended the config with the different possibilities I found:
* Either a simple setup with no proper lookup
* A proper lookup because the username is send as E-Mail
* And allowing both (if different mail clients are used for example)

**Review by nitnelave — APPROVED** [review]
Great, thanks!

**Pull request merged**